django/visions/views.py

```diff
@@ -0,0 +1,91 @@
+from django.views.generic import (TemplateView, DetailView, ListView, CreateView)
+from django.urls import reverse_lazy, reverse
+from django.http import HttpResponseRedirect
+from . import (models, forms)
+
+
+class VisionDetailView(DetailView):
+    """
+    Class-based view for Vision detail template
+    """
+    template_name = 'visions/vision-detail.html'
+    queryset = models.Vision.objects.filter(admin_approved=True)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # Add form for creating a Response object to this Vision object
+        context['response_create_form'] = forms.ResponseCreateForm
+        # Add list of responses that relate to the current vision and have been approved by admin
+        context['responses'] = models.Response.objects.filter(vision=self.object.id, admin_approved=True)
+        return context
+
+
+class VisionListView(ListView):
+    """
+    Class-based view for Vision list template
+    """
+    template_name = 'visions/vision-list.html'
+    queryset = models.Vision.objects.filter(admin_approved=True)
+    paginate_by = 100
```

Comment:
Just double-checking the pagination, as I have seen 30, and 50 previously.

Reply:
Yeah this is fine - the public interface can show a different amount to the admin interface